binary-encode.go

```diff
@@ -84,12 +84,14 @@ func (cdc *Codec) encodeReflectBinary(w io.Writer, info *TypeInfo, rv reflect.Va
 
 	case reflect.Int64:
 		if fopts.BinVarint {
-			err = EncodeVarint(w, rv.Int())
+			err = EncodeVarint(w, uint64(rv.Int()))
```

Comment:
We can keep the code as is, EncodeVarint() is the encoding for Proto3 signed varints (field type is sint64 and sint32).